Safety/boardfiles/Src/freertos.c

```diff
@@ -0,0 +1,204 @@
+
+/* USER CODE BEGIN GET_IDLE_TASK_MEMORY */
+static StaticTask_t xIdleTaskTCBBuffer;
+static StackType_t xIdleStack[configMINIMAL_STACK_SIZE];
+
+void vApplicationGetIdleTaskMemory( StaticTask_t **ppxIdleTaskTCBBuffer, StackType_t **ppxIdleTaskStackBuffer, uint32_t *pulIdleTaskStackSize )
+{
+  *ppxIdleTaskTCBBuffer = &xIdleTaskTCBBuffer;
+  *ppxIdleTaskStackBuffer = &xIdleStack[0];
+  *pulIdleTaskStackSize = configMINIMAL_STACK_SIZE;
+  /* place for user code */
+}
+/* USER CODE END FunctionPrototypes */
+
+/* Hook prototypes */
+void vApplicationStackOverflowHook(xTaskHandle xTask, signed char *pcTaskName);
+
+/* USER CODE BEGIN 4 */
+__weak void vApplicationStackOverflowHook(xTaskHandle xTask, signed char *pcTaskName)
+{
+   /* Run time stack overflow checking is performed if
+   configCHECK_FOR_STACK_OVERFLOW is defined to 1 or 2. This hook function is
+   called if a stack overflow is detected. */
+  while (1) {
+    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
+    HAL_GPIO_WritePin(LED2_GPIO_Port, LED2_Pin, GPIO_PIN_SET);
+    for (int i = 0; i < 100000; i++) { }
+    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_RESET);
+    HAL_GPIO_WritePin(LED2_GPIO_Port, LED2_Pin, GPIO_PIN_RESET);
+    for (int i = 0; i < 900000; i++) { }
+  }
+
+}
+/* USER CODE END 4 */
+
+/* Init FreeRTOS */
+
+void MX_FREERTOS_Init(void) {
+  /* USER CODE BEGIN Init */
+
+    SensorFusionInterfaceInit();
+    AttitudeManagerInterfaceInit();
+  /* USER CODE END Init */
+
+  /* USER CODE BEGIN RTOS_MUTEX */
+  /* add mutexes, ... */
+  /* USER CODE END RTOS_MUTEX */
+
+  /* USER CODE BEGIN RTOS_SEMAPHORES */
+  /* add semaphores, ... */
+  /* USER CODE END RTOS_SEMAPHORES */
+
+  /* USER CODE BEGIN RTOS_TIMERS */
+  /* start timers, add new ones, ... */
+  /* USER CODE END RTOS_TIMERS */
+
+  /* Create the thread(s) */
+  /* definition and creation of attitudeManager */
+  osThreadDef(attitudeManager, attitudeManagerExecute, osPriorityNormal, 0, 256);
+  attitudeManagerHandle = osThreadCreate(osThread(attitudeManager), NULL);
+
+  /* definition and creation of sensorFusion */
+  osThreadDef(sensorFusion, sensorFusionExecute, osPriorityNormal, 0, 2000);
+  sensorFusionHandle = osThreadCreate(osThread(sensorFusion), NULL);
+  
+  /* USER CODE BEGIN RTOS_THREADS */
+  /* add threads, ... */
+  /* USER CODE END RTOS_THREADS */
+
+  /* USER CODE BEGIN RTOS_QUEUES */
+  /* add queues, ... */
+  /* USER CODE END RTOS_QUEUES */
+}
+
+/* attitudeManagerExecute function */
+void attitudeManagerExecute(void const * argument)
+{
+
+  /* USER CODE BEGIN attitudeManagerExecute */
+  UBaseType_t uxHighWaterMark = uxTaskGetStackHighWaterMark( NULL );
+  /* Infinite loop */
+  for(;;)
+  {
+    // TODO: Re-add RSSI_CHECK
+    // RSSI_Check();
+    AttitudeManagerInterfaceExecute();
+    // HAL_GPIO_TogglePin(LED1_GPIO_Port, LED1_Pin);
+    uxHighWaterMark = uxTaskGetStackHighWaterMark( NULL );
```

Comment:
Is this to see the maximum stack usage?

Reply:
Yes. Can be removed when we don't need to make these measurements any more?